pkg/controller/eventinstance/eventinstance.go

```diff
@@ -0,0 +1,91 @@
+package eventinstance
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	v1 "github.com/acorn-io/acorn/pkg/apis/internal.acorn.io/v1"
+	"github.com/acorn-io/acorn/pkg/config"
+	"github.com/acorn-io/baaah/pkg/router"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const defaultTTL = 24 * time.Hour
+
+func GCExpired() router.HandlerFunc {
+	// last stores pre-parsed TTLs from the configuration.
+	var last struct {
+		sync.Mutex
+		raw string
+		ttl time.Duration
+	}
+
+	return handler{
+		ttl: func(ctx context.Context, getter kclient.Reader) (time.Duration, error) {
+			cfg, err := config.Get(ctx, getter)
+			if err != nil {
+				return 0, err
+			}
+
+			cfgTTL := cfg.EventTTL
+			if cfgTTL == nil || *cfgTTL == "" {
+				return defaultTTL, nil
+			}
+
+			last.Lock()
+			defer last.Unlock()
+
+			if last.raw != *cfgTTL {
+				// This is a new TTL, parse and memoize
+				ttl, err := time.ParseDuration(*cfgTTL)
+				if err != nil {
+					return 0, err
```

Comment:
This implies that, as long as there's a bad TTL in the config, expired events won't get cleaned up. Is that an acceptable failure mode?

client-side validation should make it uncommon, but the edge case will always exist as long as there's no server-side validation.

Reply:
yeah, i think so. you also ultimately log the error, so 👍

Reply:
I wonder if we can just apply a default TTL setting if the config is bad, just to protect the system from being overwhelmed

Reply:
> I wonder if we can just apply a default TTL setting if the config is bad, just to protect the system from being overwhelmed

this was the other route I was going with, but the UX feels a bit odd to me when I:
1. Set a _valid_ non-default TTL
2. Set an invalid TTL (14d "not actually valid here")
3. Am ignorant of my mistake
4. Only ever see the latest 7 days of events.